Components/Helper/Currency.php

```diff
@@ -0,0 +1,262 @@
+    const CURRENCY_SYMBOL_RIGHT = 16;
+    const CURRENCY_SYMBOL_DEFAULT = 0;
+    const CURRENCY_DECIMAL_CHAR = ',';
+    const CURRENCY_GROUPING_CHAR = '.';
+    const CURRENCY_DECIMAL_PRECISION = 2;
+
+    /**
+     * Update exchange rates for the given shop.
+     *
+     * @param Account $account
+     * @param Shop $shop
+     * @return bool
+     */
+    public static function updateCurrencyExchangeRates(Account $account, Shop $shop)
+    {
+        $currencyService = new \Nosto\Operation\SyncRates($account);
+        $collection = self::buildExchangeRatesCollection($shop);
+        try {
+            return $currencyService->update($collection);
+        } catch (\Exception $e) {
+            /** @noinspection PhpUndefinedMethodInspection */
+            Shopware()->Plugins()->Frontend()->NostoTagging()->getLogger()->error(
+                'Failed to update exchange rates: '.
+                $e->getMessage()
+            );
+            return false;
+        }
+    }
+
+    /**
+     * Get currencies for the given shop.
+     * If it's a sub shop, returns the parent's currencies
+     *
+     * @param Shop $shop
+     * @return \Doctrine\Common\Collections\ArrayCollection|Currency[]
+     */
+    public static function getCurrencies(Shop $shop)
+    {
+        $currencies = $shop->getCurrencies();
+        if (!$currencies) {
+            try {
+                // If it's a subshop, currencies are inherited from the main shop
+                $currencies = $shop->getMain()->getCurrencies();
+            } catch (\Exception $e) {
+                /** @noinspection PhpUndefinedMethodInspection */
+                Shopware()->Plugins()->Frontend()->NostoTagging()->getLogger()->warning(
+                    'Main shop has no currencies ' .
+                    $e->getMessage()
+                );
+            }
+        }
+        return $currencies;
+    }
+
+    /**
+     * @param Shop $shop
+     * @return ExchangeRateCollection
+     */
+    public static function buildExchangeRatesCollection(Shop $shop)
+    {
+        $collection = new ExchangeRateCollection();
+        foreach (self::getCurrencies($shop) as $currency) {
+            $rate = new ExchangeRate($currency->getCurrency(), (string)$currency->getFactor());
+            $collection->addRate($currency->getCurrency(), $rate);
+        }
+        return $collection;
+    }
+
+    /**
+     * @param Shop $shop
+     * @return string
+     */
+    public static function getCurrencyCode(Shop $shop)
+    {
+        if (self::isMultiCurrencyEnabled()) {
+            // Return currency code
+            $defaultCurrency = self::getDefaultCurrency();
+            if ($defaultCurrency) {
+                return $defaultCurrency->getCurrency();
+            }
+        }
+        return $shop->getCurrency()->getCurrency();
+    }
+
+    /**
+     * @return mixed|null|\Shopware\Models\Shop\Currency
+     */
+    public static function getDefaultCurrency()
```

Comment:
The logic here looks a bit scary. Does Shopware really have a installation wide default currency? Or should this be more like `getBaseCurrency`?

Reply:
That was actually getting system wide. Now it returns the default currency for the given shop, or its main shop.
Shopware is structured as `n shops` -> `n sub/subshops` AND/OR `n languageShops`
I'll leave named as `getDefaultCurrency` so it matches Shopware's naming to avoid confusion in the future.